### PR TITLE
HCAP-699/HCAP-753 withdrawal tab is available on site details view

### PR DESCRIPTION
**Pull request by Zenen-freshworks**
When a participant has been withdrawn from the program after being hired, they will be viewable under the "Withdrawn Participants" tab on the Site Details view.

### CODE REVIEW

**Review by fwpushan — CHANGES_REQUESTED** [review]
I will suggest you 
1. Use same Context, state and reducer for tab selection like Participant table
2. You can use url parameter to save selected tab state (to persist user selection between page refresh)
We can discuss on this.

**Comment by Zenen-freshworks** [comment]
Screen Shot 2021-07-14 at 2 42 00 PM
Screen Shot 2021-07-14 at 2 42 12 PM

SiteParticipantsTable has been refactored with a TabContext to hold data. Site Details have been added as a tab rather than basic display info.

File: client/src/providers/TabContext.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useReducer } from 'react';
+import { tabColumns } from '../constants';
+
+const TabContext = React.createContext();
```
Comment (fwpushan) [inline client/src/providers/TabContext.js:4]: TabContext is very generic, I will suggest a specific name like SiteDetailViewTabContext / SiteDetailTabContext

File: client/src/providers/TabContext.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    default:
+      return state;
+  }
+};
+
+/**
+ * For more: https://kentcdodds.com/blog/how-to-use-react-context-effectively
+ */
+
+const TabProvider = ({ selectedTab, siteId, children }) => {
```
Comment (fwpushan) [inline client/src/providers/TabContext.js:59]: Also same naming suggestion as TabContext

File: client/src/providers/index.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
 export * from './ToastProvider';
 export * as AuthContext from './AuthContext';
 export * as ParticipantsContext from './ParticipantsContext';
+export * as TabContext from './TabContext';
```
Comment (fwpushan) [inline client/src/providers/index.js:4]: Same naming suggestion

**Review by fwpushan — COMMENTED** [review]
Great job. All looks good, I have some naming suggestion. Please check comments.

**Review by fwpushan — DISMISSED** [review]
Nice job

**Review by dbayly-freshworks — APPROVED** [review]

**Review by dbayly-freshworks — APPROVED** [review]

**Review by fwpushan — APPROVED** [review]

**Pull request merged**